### PR TITLE
MLE-4107 Can now view MarkLogic information in a sidebar activated by the activity bar

### PR DESCRIPTION
**Pull request by BillFarber**

### CODE REVIEW

**Comment by BillFarber** [comment]
I realized that I have done anything about potential errors, so I've converted this to a draft PR for the time-being.

File: client/extension.ts [hunk 1]
```diff
@@ -119,28 +121,36 @@ export function activate(context: vscode.ExtensionContext): void {
         const cfg: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration();
         const dbClientContext: ClientContext = getDbClientWithoutOverrides(cfg, context.globalState);
         if (dbClientContext) {
-            JsDebugManager.connectToJsDebugServer(dbClientContext);
+            JsDebugManager.connectToJsDebugServer(dbClientContext).then(() => {
+                vscode.commands.executeCommand('markLogicServerStatus.refreshEntry');
```
Comment (rjrudin) [inline client/extension.ts]: This line looks like a function waiting to be extracted, as it appears to be the same in several places - e.g. `refreshServerStatus` (though I'm wondering if it should reference the actual GUI component, which I think would be more self-documenting - e.g. `refreshServerStatusInActivityBar`?).

File: client/extension.ts [hunk 2]
```diff
@@ -230,6 +240,11 @@ export function activate(context: vscode.ExtensionContext): void {
     JsDebugManager.registerMlxprsStatusBarItem(mlxprsStatus);
     mlxprsStatus.requestUpdate();
 
+    // Add a MarkLogic ActivityBar entry
```
Comment (rjrudin) [inline client/extension.ts]: I'm wondering about the naming here. The server status is shown in a VSCode "tree data provider", so the variable name below makes sense. Does calling `registerTreeDataProvider` automatically add it to what VSCode calls the "activity bar"?

File: client/marklogicClient.ts [hunk 3]
```diff
@@ -221,6 +245,63 @@ return $json-servers
...
+                .result((result: ServerListResponse) => {
+                    resolve(result);
+                })
+                .catch((error) => {
+                    reject(error);
+                });
+        });
+    }
+
+    async executeGenericGetRequest(url: string): Promise<string> {
```
Comment (rjrudin) [inline client/marklogicClient.ts]: Did you mean to reuse this above? I saw the duplication in getDatabaseList and getAppServerList and was thinking - looks like there's a generic GET function to be extracted that just accepts a path - and that seems to be this function.

File: client/marklogicServerStatus.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        ];
+    }
+
+    private async buildDatabaseEntries(): Promise<MarkLogicServerStatus[]> {
+        return this.dbManageClientContext.getDatabaseList()
+            .then((databaseResponse) => {
+                const serverStatusList: MarkLogicServerStatus[] = [];
+                const responseList = databaseResponse['database-default-list']['list-items']['list-item'];
+                responseList.forEach(databaseItem => {
+                    const url = `http://localhost:8001/database-admin.xqy?section=database&database=${databaseItem.idref}`;
```
Comment (rjrudin) [inline client/marklogicServerStatus.ts]: This URL is hardcoded, and presumably can have the scheme, host, and post all replaced (I forget if mlxprs supports SSL, but assuming it does, you'd need to determine if you want "http" or "https" here).

File: client/extension.ts (hunk 1 above)
Comment (BillFarber) [reply client/extension.ts]: Are you talking about all 3 of these lines repeating?
```
            JsDebugManager.connectToJsDebugServer(dbClientContext).then(() => {
                vscode.commands.executeCommand('markLogicServerStatus.refreshEntry');
            });
```
If so, they don't quite repeat. That first line is a bit different each time.

**Review by rjrudin — APPROVED** [review]
Looks good!

**Pull request merged**